easybuild/easyconfigs/m/MFiX/MFiX-24.3-intel-2023b.eb

```
@@ -0,0 +1,55 @@
+# Author: Ehsan Moravveji (VSC, KU Leuven)
+
+easyblock = 'CMakeMake'
+
+name = 'MFiX'
+version = '24.3'
+
+homepage = 'https://mfix.netl.doe.gov/products/mfix/'
+description = """
+The National Energy Technology Laboratory’s (NETL’s) computational fluid dynamics (CFD) code, 
+MFiX-Multiphase Flow with Interphase eXchanges - is central to the laboratory’s multiphase flow reactor 
+modeling efforts. This open-source software has over three decades of development history and more than 7,000 
+registered users worldwide.  MFiX has become the standard for comparing, implementing, and evaluating multiphase 
+flow constitutive models and has been applied to an extremely diverse range of multiphase flows applications. 
+The successes achieved in modeling complex multiphase flow systems have led to new and improved key attributes 
+such as drag, polydispersity, attrition, and agglomeration models, among other significant advances.
+"""
+
+toolchain = {'name': 'intel', 'version': '2023b'}
+toolchainopts = {
+    'usempi': True,
+    'pic': True,
+    'extra_fflags': '-fallow-argument-mismatch',
+}
+
+sources = ['%(namelower)s-%(version)s.tar.gz']
+checksums = ['d6313d02b4631a6b2eef944c158bb7b1a2eecfe7bcee9f533ce71571f1acffe8']
+
+download_instructions = 'Fetch the source %s after logging into %s' % (sources[0], homepage)
+
+builddependencies = [
+    ('CMake', '3.27.6'),
+    ('binutils', '2.40'),
+]
+
+configopts = '-DENABLE_MPI=1'
+
+separate_build_dir = False
+
+test_cmd = ' && '.join([
+    'cd tutorials/dem/mixer_3d',
+    '%(builddir)s/%(namelower)s-%(version)s/mfixsolver_dmp -f mixer_dem_3d.mfx',
+])
```

Without seeing the sources I can't be sure, but I believe the package has ctest support (and OpenMP support) which can be enabled via configopts. if that works then you should be able to run
```suggestion
test_cmd = ' && '.join([
    'ctest',
    'cd tutorials/dem/mixer_3d',
    '%(builddir)s/%(namelower)s-%(version)s/mfixsolver_dmp -f mixer_dem_3d.mfx',
])
```

---

I tried to enable the `ctest` by passing the `-DENABLE_CTEST` (which also required `numdiff` module as an additional dependency), but then things blew up immediately at `cmake` step. Hence, I took it out.

---

I'd be interested to know what "blew up" looks like, if a package ships a test suite it is always better to try to run it.

---

Here is a tarball with a modified easyconfig which enables ctest and the resulting logfile.
[reprod_ctest.tar.gz](https://github.com/user-attachments/files/18575397/reprod_ctest.tar.gz)

---

Ok, yeah, certainly looks like you don't have the complete sources to be able to enable tests